include/envoy/http/header_map.h

```diff
@@ -22,6 +22,10 @@ class LowerCaseString {
 
   const std::string& get() const { return string_; }
   bool operator==(const LowerCaseString& rhs) const { return string_ == rhs.string_; }
+  LowerCaseString& operator=(const LowerCaseString& rhs) {
```

Comment:
that's because Optional does not have a support for moveable types. Would you suggest to add it there?